### DISCUSSION TITLE
Invalid Character data was found - but only sometimes

### DISCUSSION THREAD

**Discussion by richarddemke-lgcy** [dbeaver/dbeaver#17073] [General]
#### System information: 
- Windows 11
- DBeaver version 22.1.1

#### Connection specification:
- Postgres 14.2 
- postgres-jdbc (Dbeaver Auto Installed)
- Using AWS RDS

#### Describe the problem you're observing:
I can't wrap my head around it. I have a MySQL_FDW for some older databases. We are consolidating into Postgres. The MySQL Database is Utf8Mb4 and the Postgres Db is utf8. Not doing anything fancy. When I query against the foreign MySQL table, I sometimes get following error (i will send the entire log): 

org.jkiss.dbeaver.model.exec.DBCException: SQL Error [22000]: Invalid character data was found.  This is most likely caused by stored data containing characters that are invalid for the character set the database was created in.  The most common example of this is storing 8bit data in a SQL_ASCII database.
Caused by: org.postgresql.util.PSQLException: Invalid character data was found.  This is most likely caused by stored data containing characters that are invalid for the character set the database was created in.  The most common example of this is storing 8bit data in a SQL_ASCII database.
Caused by: java.io.IOException: Illegal UTF-8 sequence: byte 2 of 3 byte sequence is not 10xxxxxx: 110

Again- my server and database are Utf8, not SQL_ASCII. 

I began troubleshooting and used to_ascii(), unaccent(), regex_replace, etc. to remove these values. I somehow got it to work- so I thought. I saw the error again after I swear I had fixed it. What's odd is sometimes it occurs and sometimes it does not. I have two different queries, a simple select on the field in question (the field is customer_name):

Select id, customer_name
from accounts
where id = 199063
-- there are others

If I wait a few minutes after running it several times, the query will magically work and the name will display with all of its accented characters. I am providing two pictures of the same query and result set. 

I am going nuts. This has caused some issues with our API as the payloads fail. I am at a loss. 
[dbeaver log.txt](https://github.com/dbeaver/dbeaver/files/9073484/dbeaver.log.txt)
![dbeaver query success](https://user-images.githubusercontent.com/99755790/178030305-d201cbad-178f-4d8c-97d2-f0826888af47.png)
![dbeaver query fail](https://user-images.githubusercontent.com/99755790/178030306-57115930-b0ae-4ca1-ab4c-7d6d1d9de712.png)
[dbeaver log.txt](https://github.com/dbeaver/dbeaver/files/9073490/dbeaver.log.txt)

**Comment by emironovDB** [comment]
Hello @richarddemke-lgcy,

If I understand correctly, you're accessing an MySQL database through PostgreSQL.

Looks like that MySQL database has encoding that is not UTF-8 (maybe there's something with column's collation).

It doesn't seem to be DBeaver issue.

**Reply by richarddemke-lgcy** [reply]
I managed both databases. The MySQL database is definitely UTF8 and the column is as well (utf8mb4 to be exact). 

What doesn't make sense it why the same query being executed display one result whilst the other errors out.

**Reply by richarddemke-lgcy** [reply]
@emironovDB
